src/sourcemaps/tools/sentry-cli.ts

```diff
@@ -0,0 +1,120 @@
+// @ts-ignore - clack is ESM and TS complains about that. It works though
+import clack from '@clack/prompts';
+import chalk from 'chalk';
+import * as path from 'path';
+import * as fs from 'fs';
+import {
+  abortIfCancelled,
+  addSentryCliRc,
+  getPackageDotJson,
+  hasPackageInstalled,
+  installPackage,
+} from '../../utils/clack-utils';
+
+import { SourceMapUploadToolConfigurationFunction } from './types';
+
+export const configureSentryCLI: SourceMapUploadToolConfigurationFunction =
+  async (options) => {
+    const packageDotJson = await getPackageDotJson();
+
+    await installPackage({
+      packageName: '@sentry/cli',
+      alreadyInstalled: hasPackageInstalled('@sentry/cli', packageDotJson),
+    });
+
+    let validPath = false;
+    let relativeArtifactPath;
+    do {
+      const rawArtifactPath = await abortIfCancelled(
+        clack.text({
+          message:
+            'Provide a path to the folder that contains your build artifacts:',
+          placeholder: `path${path.sep}to${path.sep}your${path.sep}build${path.sep}output`,
+        }),
+      );
+
+      if (path.isAbsolute(rawArtifactPath)) {
+        relativeArtifactPath = path.relative(process.cwd(), rawArtifactPath);
+      } else {
+        relativeArtifactPath = rawArtifactPath;
+      }
+
+      try {
+        await fs.promises.access(
+          path.join(process.cwd(), relativeArtifactPath),
+        );
+        validPath = true;
+      } catch {
+        validPath = await abortIfCancelled(
+          clack.select({
+            message: `We couldn't find artifacts at ${relativeArtifactPath}. Are you sure that this is the location that contains your build artifacts?`,
+            options: [
+              {
+                label: 'No, let me verify.',
+                value: false,
+              },
+              { label: 'Yes, I am sure!', value: true },
+            ],
+            initialValue: false,
+          }),
+        );
+      }
+    } while (!validPath);
+
+    const relativePosixArtifactPath = relativeArtifactPath
+      .split(path.sep)
+      .join(path.posix.sep);
+
+    await abortIfCancelled(
+      clack.select({
+        message: `Verify that your build tool is generating source maps. ${chalk.dim(
+          '(Your build output folder should contain .js.map files after a build)',
+        )}`,
+        options: [{ label: 'I checked. Continue!', value: true }],
+        initialValue: true,
+      }),
+    );
+
+    packageDotJson.scripts = packageDotJson.scripts || {};
+    packageDotJson.scripts[
+      'sentry:ci'
+    ] = `sentry-cli sourcemaps inject --org ${options.orgSlug} --project ${options.projectSlug} ${relativePosixArtifactPath} && sentry-cli sourcemaps upload --org ${options.orgSlug} --project ${options.projectSlug} ${relativePosixArtifactPath}`;
```

Comment:
m: We need to pass the `url` here, too, for self-hosted, right?

Reply:
ah right